code/game/objects/items/sharpener.dm

```diff
@@ -1,57 +1,77 @@
-
+	///If TRUE, the whetstone will only sharpen already sharp items
+	var/requires_sharpness = TRUE
 
 /obj/item/sharpener/attackby(obj/item/I, mob/user, params)
-	if(used)
+	if(used) //You can't sharpen stuff with a worn whetstone
 		to_chat(user, "<span class='warning'>The sharpening block is too worn to use again!</span>")
 		return
-	if(I.force >= max || I.throwforce >= max)//no esword sharpening
+	if(I.force >= max || I.throwforce >= max) //If we're trying to sharpen an item that already has a force higher than max, we fail
 		to_chat(user, "<span class='warning'>[I] is much too powerful to sharpen further!</span>")
 		return
-	if(requires_sharpness && !I.get_sharpness())
+	if(requires_sharpness && !I.get_sharpness()) //You can't sharpen an item that isn't sharp with a whetstone that requieres items to be sharp
 		to_chat(user, "<span class='warning'>You can only sharpen items that are already sharp, such as knives!</span>")
 		return
-	if(is_type_in_list(I, list(/obj/item/melee/transforming/energy, /obj/item/dualsaber)))
+	if(is_type_in_list(I, list(/obj/item/melee/transforming/energy, /obj/item/dualsaber))) //You can't sharpen photons
 		to_chat(user, "<span class='warning'>You don't think \the [I] will be the thing getting modified if you use it on \the [src]!</span>")
 		return
 
-	var/signal_out = SEND_SIGNAL(I, COMSIG_ITEM_SHARPEN_ACT, increment, max)
-	if(signal_out & COMPONENT_BLOCK_SHARPEN_MAXED)
+	var/signal_out = SEND_SIGNAL(I, COMSIG_ITEM_SHARPEN_ACT, increment, max) //This is used to check more things if the item has a relevant component. As of December 2020, this is only used for two_handed and transforming.
+	if(signal_out & COMPONENT_BLOCK_SHARPEN_MAXED) //If the item's components enforce more limits on maximum power from sharpening,  we fail
 		to_chat(user, "<span class='warning'>[I] is much too powerful to sharpen further!</span>")
 		return
-	if(signal_out & COMPONENT_BLOCK_SHARPEN_BLOCKED)
+	if(signal_out & COMPONENT_BLOCK_SHARPEN_BLOCKED) //Checks for "other" restrictions
 		to_chat(user, "<span class='warning'>[I] is not able to be sharpened right now!</span>")
 		return
-	if((signal_out & COMPONENT_BLOCK_SHARPEN_ALREADY) || (I.force > initial(I.force) && !signal_out))
+	if((signal_out & COMPONENT_BLOCK_SHARPEN_ALREADY) || (I.force > initial(I.force) && !signal_out)) //No sharpening stuff twice
 		to_chat(user, "<span class='warning'>[I] has already been refined before. It cannot be sharpened further!</span>")
 		return
-	if(!(signal_out & COMPONENT_BLOCK_SHARPEN_APPLIED))
-		I.force = clamp(I.force + increment, 0, max)
-		I.wound_bonus = I.wound_bonus + increment
+	if(!(signal_out & COMPONENT_BLOCK_SHARPEN_APPLIED)) //If the item has a relevant component and COMPONENT_BLOCK_SHARPEN_APPLIED is returned, the item only gets the throw force increase
+		I.force = clamp(I.force + increment, 0, max) //We can't end up with a force better than max
+		I.wound_bonus = I.wound_bonus + increment //wound_bonus has no cap
```

Comment:
These are another instance of overdetail, tho the last if statement is a bit better, since there's other info there too. Marking wound_bonus as an issue is also v good